runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,417 @@
+      return
+    elseif line:find("^%s*;") then
+      vim.bo[bufnr].filetype = "msmessages"
+      return
+    end
+  end
+  -- Default: Sendmail .mc file
+  vim.bo[bufnr].filetype = "m4"
+end
+
+function M.mm(path, bufnr) end
+
+function M.mms(bufnr)
+  for _, line in ipairs(getlines(bufnr, 1, 20)) do
+    if findany(line, { "^%s*%%", "^%s*//", "^%*" }) then
+      vim.bo[bufnr].filetype = "mmix"
+      return
+    elseif line:find("^%s*#") then
+      vim.bo[bufnr].filetype = "make"
+      return
+    end
+  end
+  vim.bo[bufnr].filetype = "mmix"
+end
+
+function M.mod(path, bufnr) end
+
+-- This function checks if one of the first five lines start with a dot. In
+-- that case it is probably an nroff file: 'filetype' is set and 1 is returned.
+function M.nroff(bufnr)
+  for _, line in ipairs(getlines(bufnr, 1, 5)) do
+    if line:find("^%.") then
+      vim.bo[bufnr].filetype = "nroff"
+      return 1
+    end
+  end
+  return 0
+end
+
+function M.perl(path, bufnr) end
+
+function M.pl(path, bufnr) end
+
+function M.pp(path, bufnr) end
+
+function M.prg(path, bufnr) end
+
+function M.progress_asm(path, bufnr) end
+
+function M.progress_cweb(bufnr)
+  if vim.g.filetype_w then
+    vim.bo[bufnr].filetype = vim.g.filetype_w
+  else
+    if getlines(bufnr, 1):find("^&ANALYZE") or getlines(bufnr, 3):find("^&GLOBAL%-DEFINE") then
+      vim.bo[bufnr].filetype = "progress"
+    else
+      vim.bo[bufnr].filetype = "cweb"
+    end
+  end
+end
+
+function M.progress_pascal(path, bufnr) end
+
+function M.proto(path, bufnr) end
+
+function M.r(bufnr)
+  local lines = getlines(bufnr, 1, 50)
+  -- TODO: \< / \> which match the beginning / end of a word
+  -- Rebol is easy to recognize, check for that first
+  if table.concat(lines):lower():find("rebol") then
+    vim.bo[bufnr].filetype = "rebol"
+    return
+  end
+
+  for _, line in ipairs(lines) do
+    -- R has # comments
+    if line:find("^%s*#") then
+      vim.bo[bufnr].filetype = "r"
+      return
+    end
+    -- Rexx has /* comments */
+    if line:find("^%s*/%*") then
+      vim.bo[bufnr].filetype = "rexx"
+      return
+    end
+  end
+
+  -- Nothing recognized, use user default or assume R
+  if vim.g.filetype_r then
+    vim.bo[bufnr].filetype = vim.g.filetype_r
+  else
+    -- Rexx used to be the default, but R appears to be much more popular.
+    vim.bo[bufnr].filetype = "r"
+  end
+end
+
+function M.redif(bufnr)
+  for _, line in ipairs(getlines(bufnr, 1, 5)) do
+    if line:lower():find("^template%-type:") then
+      vim.bo[bufnr].filetype = "redif"
+    end
+  end
+end
+
+function M.rules(path, bufnr) end
+
+-- This function checks the first 25 lines of file extension "sc" to resolve
+-- detection between scala and SuperCollider
+function M.sc(bufnr)
+  for _, line in ipairs(getlines(bufnr, 1, 25)) do
+    if
+      findany(
+        line,
+        { "[A-Za-z0-9]*%s:%s[A-Za-z0-9]", "var%s<", "classvar%s<", "^this.*", "|%w*|", "%+%s%w*%s{", "%*ar%s" }
```

Comment:
Test indicates that something here is not equivalent to the vim version; it's failing to recognize
```
// Query
//Method
^this {
```
as `supercollider`

I think this is meant to be a _literal_ `^`?